diff --git a/components/pango_opengl/include/pangolin/gl/glplatform.h b/components/pango_opengl/include/pangolin/gl/glplatform.h
index 94ea1baf..38d06654 100644
--- a/components/pango_opengl/include/pangolin/gl/glplatform.h
+++ b/components/pango_opengl/include/pangolin/gl/glplatform.h
@@ -79,8 +79,9 @@
         #define GL_SILENCE_DEPRECATION
         #include <OpenGL/gl.h>
     #else
-        #include <GL/gl.h>
+        //#include <GL/gl.h>
     #endif
 #endif // HAVE_GLES
+#include <GL/glew.h>
 
 #include <pangolin/gl/glpangoglu.h>
