diff --git a/CMakeLists.txt b/CMakeLists.txt
index dbdbdb4..79f4f33 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,7 +80,6 @@ endif(WIN32)
 add_library(${PROJECT_NAME} ${LIB_SHARED} ${SRCS})
 target_link_libraries(${PROJECT_NAME} ${OpenCV_LIBS})
 target_include_directories(${PROJECT_NAME} PUBLIC include/DUtils include/DUtilsCV include/DVision ${OpenCV_INCLUDE_DIRS})
-set_target_properties(${PROJECT_NAME} PROPERTIES CXX_STANDARD 11)
 if (Boost_FOUND)
   target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
   target_include_directories(${PROJECT_NAME} PUBLIC ${Boost_INCLUDE_DIRS})
